src/gui/layout/qgslayoutlabelwidget.cpp

```diff
@@ -146,7 +146,7 @@ void QgsLayoutLabelWidget::buildInsertDynamicTextMenu( QgsLayout *layout, QMenu
     QMenu *mapMenu = new QMenu( map->displayName(), mapsMenu );
     for ( const std::pair< QString, QString > &expression :
           {
-            std::make_pair( tr( "Scale (%1)" ).arg( map->scale() ), QStringLiteral( "item_variables('%1')['map_scale']" ).arg( map->id() ) ),
+            std::make_pair( tr( "Scale (%1)" ).arg( map->scale() ), QStringLiteral( "format_number(item_variables('%1')['map_scale'], 6, omit_group_separators:=true, trim_trailing_zeroes:=true)" ).arg( map->id() ) ),
```

Comment:
I'm curious, why 6 decimals here?

Reply:
I figure it can't hurt to go more vs less given we are trimming trailing zeroes, and there's at least report in the system from users who are using print layouts for micro (macro?) scale work in the sub-mm range.

Reply:
@nyalldawson , if we'd be willing to complexify the formula, would could make the nb of decimal based on scale (i.e. < if (scale < 100, 6, 3)).

Reply:
I'm reluctant to make these too complex, as they are a nice way to gently expose users to the expression engine. If they get too fancy then we'll lose the transparency of what's actually happening. 

What about if we added the argument name for the decimals parameter so that it's a bit clearer to a user what this number means (and hopefully empower them to easily change it themselves)?

Reply:
@nyalldawson , sounds good.

Reply:
> I'm reluctant to make these too complex, as they are a nice way to gently expose users to the expression engine. If they get too fancy then we'll lose the transparency of what's actually happening.
> 
> What about if we added the argument name for the decimals parameter so that it's a bit clearer to a user what this number means (and hopefully empower them to easily change it themselves)?

Exposing the number would be a good idea, though what lead to the ticket was the difference between the two displayed values. I'm guessing having the same formatting would make sense in this case as this is show as a way to display outwark what is in the UI.

Though what might be more interesting is exposing a variable that countrols the level or rounding above the decimals (decimal precision in scale should be irrelevant in nearly all cases) but having a way to round to the 1000s or 100s might be a nice things to let users tweak and get them more at ease with expressions. Should format number accept negative values to allow this kind of rounding?

Reply:
> Should format number accept negative values to allow this kind of rounding?

Sounds like a great idea!